backend/backend/services/graph/subgraphs.py

```diff
@@ -9,63 +9,57 @@
-            try:
+import yaml
 
-                schema_file = os.path.join(os.getcwdb().decode("utf-8") + PATH, protocol, "schema.graphql")
-                with open(schema_file) as f:
-                    first_line = f.readline()
-                    protocols[protocol]["schema_file"] = schema_file
-                # gather deployment data for protocol
-                if protocol in deployment:
-                    protocols[protocol]["type"] = deployment[protocol]["schema"]
-                    protocols[protocol]["deployments"] = {}  # network_label: {}}
-                    for chain in deployment[protocol]["deployments"]:
 
-                        # if deployment[protocol]["deployments"][chain]["status"] == "prod":
-                        network_label = deployment[protocol]["deployments"][chain][
-                            "network"
-                        ]
-                        protocols[protocol]["deployments"][network_label] = deployment[
-                            protocol
-                        ]["deployments"][chain]["services"]
-                        if len(protocols[protocol]["deployments"]) == 0:
-                            # no production ready deployments founds
-                            unfinished_protocols.append(protocol)
-                if "deployments" not in protocols[protocol]:
-                    # no deployments founds
-                    unfinished_protocols.append(protocol)
-            except FileNotFoundError:
-                # no graphql schema found
-                unfinished_protocols.append(protocol)
-        # for protocol in set(unfinished_protocols):
-        #     protocols.pop(protocol)
-
-
-        return protocols
+class SubgraphService:
+    def __init__(self, protocol, chain):
+        self.protocol = protocol
+        self.chain = chain
+        self.deployments = json.load(
+            open(os.getcwdb().decode("utf-8") + "/subgraphs/deployment/deployment.json")
+        )[protocol]
+        if (
+            "decentralized-network"
+            in self.deployments["deployments"][f"{protocol}-{chain}"]["services"]
+        ):
+            self.service_type = "decentralized-network"
+        elif (
+            "hosted-service"
+            in self.deployments["deployments"][f"{protocol}-{chain}"]["services"]
+        ):
+            self.service_type = "hosted-service"
+        try:
+            self.query_id = self.deployments["deployments"][f"{protocol}-{chain}"][
+                "services"
+            ][self.service_type]["query-id"]
+        except AttributeError:
+            # protocol does not have any deployments
+            raise NotImplementedError
+        try:
+            self.template_file_location = f"subgraphs/subgraphs/{self.deployments['base']}/protocols/{self.protocol}/config/templates/{self.deployments['deployments'][f'{protocol}-{chain}']['files']['template']}"
+        except:
+            # no template file defined
```

Comment:
We should still set this `self.template_file_location = None` for proper object typing

Reply:
e41910facc691dd7c5c3236ebf599763cd691973